restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,374 @@
+                    Kind.Regular,
+                    Type.fromCqlDefinitionOf(keyspace, colDef.getTypeDefinition())));
+          }
+          // Build target object from definition
+          UserDefinedType udt =
+              ImmutableUserDefinedType.builder()
+                  .keyspace(keyspaceName)
+                  .name(typeName)
+                  .addColumns((Column[]) columns.toArray(new Column[0]))
+                  .build();
+          // Execute creation
+          authenticatedDB
+              .getDataStore()
+              .queryBuilder()
+              .create()
+              .type(keyspaceName, udt)
+              .ifNotExists(udtAdd.isIfNotExists())
+              .build()
+              .execute(ConsistencyLevel.LOCAL_QUORUM)
+              .get();
+          return Response.status(Response.Status.CREATED)
+              .entity(Converters.writeResponse(Collections.singletonMap("name", typeName)))
+              .build();
+        });
+  }
+
+  @Timed
+  @DELETE
+  @ApiOperation(
+      value = "Delete a User Defined type (UDT)",
+      notes = "Delete a single user defined type (UDT) in the specified keyspace.")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 204, message = "No Content"),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  @Path("/{typeName}")
+  public Response delete(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(
+              value = "Name of the user defined type (UDT) to use for the request.",
+              required = true)
+          @PathParam("typeName")
+          final String typeName,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+
+          // Permissions on a type are the same as keyspace
+          // (CreateTypeFetcher for graphQl API)
+          db.getAuthorizationService()
+              .authorizeSchemaRead(
+                  authenticatedDB.getAuthenticationSubject(),
+                  Collections.singletonList(keyspaceName),
+                  null,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+
+          authenticatedDB
+              .getDataStore()
+              .queryBuilder()
+              .drop()
+              .type(
+                  keyspaceName,
+                  ImmutableUserDefinedType.builder().keyspace(keyspaceName).name(typeName).build())
```

Comment:
Just double checking here. There's no way this can lead to NPE if the keyspace doesn't exists, right?

Reply:
Do not see any NPE in the `QueryBuilderImpl`, still checking existence before is required as stated above